packages/@storefront/flux-capacitor/src/core/store/middleware.ts

```diff
@@ -147,6 +147,24 @@ export namespace Middleware {
     };
   }
 
+  export function redirectAnalyzer() {
+    let isRedirecting = false;
+
+    return () => (next) => (action) => {
+      if (action.type === Actions.START_REDIRECT) {
+        isRedirecting = true;
+      } else if (action.type === Actions.DONE_REDIRECT) {
+        isRedirecting = false;
+      }
+
+      if (isRedirecting) {
```

Comment:
Does this current approach mean that a `DONE_REDIRECT` action must be dispatched after a `START_REDIRECT` action in order for SF to return to normal functionality after a redirect?

Reply:
That's correct. `START_REDIRECT` causes all subsequent actions *other than* `DONE_REDIRECT` to be ignore. That said, `START_REDIRECT` is dispatched immediately after `window.location.assign()` is invoked. At some point, `assign()` *will* navigate to the target page. This will re-instantiate the application and restore the default behaviour.

Reply:
Makes sense, resolving this thread.